### PR TITLE
Enable building of System.Linq.Expression without expression compiler

### PR DESCRIPTION
**Pull request by VSadov**
Enable building of System.Linq.Expression without expression compiler in AOT target group.

Fixes:https://github.com/dotnet/corefx/issues/16910

### CODE REVIEW

**Comment by VSadov** [comment]
CC: @DnlHarvey @danmosemsft @OmarTawfik 

Not sure if this will pass tests. (not sure tests even run with AOT), but the goal is to first get this built without compiler when targeting AOT.

**Comment by danmosemsft** [comment]
LGTM but not my area obviously.

**Comment by danmosemsft** [comment]
@joperezr fyi

**Comment by VSadov** [comment]
CC: @jaredpar - FYI

File: src/System.Linq.Expressions/tests/System.Linq.Expressions.Tests.csproj [hunk 1]
```diff
@@ -8,10 +8,6 @@
   <PropertyGroup>
     <ProjectGuid>{4B4AA59B-89F9-4A34-B3C3-C97EF531EE00}</ProjectGuid>
     <IsInterpreting Condition="'$(TargetGroup)' == 'uap'">true</IsInterpreting>
-    <!-- Temporarily disable IsInterpreting build as it's broken and needs investigation
-         https://github.com/dotnet/corefx/issues/16910
-    -->
-    <IsInterpreting>false</IsInterpreting>
     <DefineConstants Condition=" '$(IsInterpreting)' != 'true' ">$(DefineConstants);FEATURE_COMPILE</DefineConstants>
```
Comment (danmosemsft) [inline src/System.Linq.Expressions/tests/System.Linq.Expressions.Tests.csproj]: What is the difference between FEATURE_COMPILE and "not FEATURE_INTERPRET" ?  Might be worth a comment?

File: src/System.Linq.Expressions/tests/TestExtensions/PerCompilationTypeAttribute.cs [hunk 2]
```diff
@@ -33,25 +33,40 @@ public override IEnumerable<object[]> GetData(MethodInfo testMethod)
...
                     withTrue[i] = arg;
+#endif
                 }
 
+#if FEATURE_COMPILE
                 yield return withFalse;
-#if FEATURE_COMPILE && FEATURE_INTERPRET
+#endif
+
+#if  FEATURE_INTERPRET
```
Comment (jaredpar) [inline src/System.Linq.Expressions/tests/TestExtensions/PerCompilationTypeAttribute.cs]: Are these conditional compiles documented anywhere?

**Comment by joperezr** [comment]
@VSadov looks like even after your changes System.Linq.Expressions still depends on System.Reflection.Emit.Lightweight which wont supported on uapaot. Is there a plan to remove that dependency? cc: @jkotas

**Pull request merged**